src/views/Header/Header.js

```diff
@@ -149,8 +150,9 @@ class Header extends React.PureComponent<Props, State> {
     const width = this.state.widths[props.scene.key]
       ? (this.props.layout.initWidth - this.state.widths[props.scene.key]) / 2
       : undefined;
+    const UsedBackButton = options.HeaderLeftComponent || HeaderBackButton;
```

Comment:
What's the difference between `HeaderLeftComponent` and `headerLeft`? I think this API would be confusing - it doesn't seem consistent with other naming conventions used.

Reply:
Thanks for the question @spencercarli! The difference is that `headerLeft` is an element, but `HeaderLeftComponent` is a component. The motivation for the change is described in the description, and the naming was inspired by react-native core, for example by [FlatList](https://github.com/facebook/react-native/blob/515eb0e8012a7a8f085a8e410c6c694011fd8c1d/Libraries/Lists/FlatList.js#L94). BUT looking at the code I see that `ListHeaderComponent?: ?(React.ComponentType<any> | React.Element<any>),`, ie. `ListHeaderComponent` can be an element or a component. 

So, let's not introduce another prop but take advantage of the existing `headerLeft`: my suggestion is that `headerLeft`, which now can be a react element, can also be a react component. Based on whether it is an element or component, we will render the appropriate thing: if it is an element, we just return it, and if it is a component, we create an element out of it, and return that. Just like [in VirtualizedList](https://github.com/facebook/react-native/blob/515eb0e8012a7a8f085a8e410c6c694011fd8c1d/Libraries/Lists/VirtualizedList.js#L752).

Reply:
I think that's a great idea, @vonovak!